File: src/main/java/io/github/mg138/tsbook/items/data/stat/map/DamageDefenseRelation.java
package io.github.mg138.tsbook.items.data.stat.map;

import io.github.mg138.tsbook.items.data.stat.StatType;

import java.util.EnumMap;

public class DamageDefenseRelation {
    private static EnumMap<StatType, StatType> constructRelationship() {
        EnumMap<StatType, StatType> relation = new EnumMap<>(StatType.class);
        relation.put(StatType.DAMAGE_PHYSICAL, StatType.DEFENSE_PHYSICAL);
        relation.put(StatType.DAMAGE_TEMPUS, StatType.DEFENSE_TEMPUS);
        relation.put(StatType.DAMAGE_TERRA, StatType.DEFENSE_TERRA);
        relation.put(StatType.DAMAGE_IGNIS, StatType.DEFENSE_IGNIS);
        relation.put(StatType.DAMAGE_AQUA, StatType.DEFENSE_AQUA);
        relation.put(StatType.DAMAGE_UMBRA, StatType.DEFENSE_UMBRA);
        relation.put(StatType.DAMAGE_LUMEN, StatType.DEFENSE_LUMEN);
        relation.put(StatType.DAMAGE_TRUE, StatType.DEFENSE_TRUE);
        return relation;
    }

    public final static EnumMap<StatType, StatType> relationship = constructRelationship();
}

File: src/main/java/io/github/mg138/tsbook/items/data/stat/util/map/DamageDefenseRelation.kt
package io.github.mg138.tsbook.items.data.stat.util.map

import io.github.mg138.tsbook.items.data.stat.StatType
import java.util.*

object DamageDefenseRelation {
    private fun constructRelationship(): EnumMap<StatType, StatType> {
        val map = EnumMap<StatType, StatType>(StatType::class.java)
        map[StatType.DAMAGE_PHYSICAL] = StatType.DEFENSE_PHYSICAL
        map[StatType.DAMAGE_TEMPUS] = StatType.DEFENSE_TEMPUS
        map[StatType.DAMAGE_TERRA] = StatType.DEFENSE_TERRA
        map[StatType.DAMAGE_IGNIS] = StatType.DEFENSE_IGNIS
        map[StatType.DAMAGE_AQUA] = StatType.DEFENSE_AQUA
        map[StatType.DAMAGE_UMBRA] = StatType.DEFENSE_UMBRA
        map[StatType.DAMAGE_LUMEN] = StatType.DEFENSE_LUMEN
        map[StatType.DAMAGE_TRUE] = StatType.DEFENSE_TRUE
        return map
    }

    val relationship = constructRelationship()
}